nsxt/resource_nsxt_policy_lb_icmp_monitor_profile.go

```diff
@@ -0,0 +1,190 @@
+/* Copyright © 2020 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtPolicyLBIcmpMonitorProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyLBIcmpMonitorProfileCreate,
+		Read:   resourceNsxtPolicyLBIcmpMonitorProfileRead,
+		Update: resourceNsxtPolicyLBIcmpMonitorProfileUpdate,
+		Delete: resourceNsxtPolicyLBIcmpMonitorProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"data_length": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The data size (in byte) of the ICMP healthcheck packet format: int64",
+			},
+			"fall_count":   getLbMonitorFallCountSchema(),
+			"interval":     getLbMonitorIntervalSchema(),
+			"monitor_port": getPolicyLbMonitorPortSchema(),
+			"rise_count":   getLbMonitorRiseCountSchema(),
+			"timeout":      getLbMonitorTimeoutSchema(),
+		},
+	}
+}
+
+func resourceNsxtPolicyLBIcmpMonitorProfileExists(id string, connector client.Connector, isGlobalManager bool) (bool, error) {
+	var err error
+	client := infra.NewLbMonitorProfilesClient(connector)
+	_, err = client.Get(id)
+	if err == nil {
+		return true, nil
+	}
+
+	if isNotFoundError(err) {
+		return false, nil
+	}
+
+	return false, logAPIError("Error retrieving resource", err)
+}
+
+func resourceNsxtPolicyLBIcmpMonitorProfilePatch(d *schema.ResourceData, m interface{}, id string) error {
+	connector := getPolicyConnector(m)
+	converter := bindings.NewTypeConverter()
+
+	displayName := d.Get("display_name").(string)
+	description := d.Get("description").(string)
+	tags := getPolicyTagsFromSchema(d)
+	dataLength := int64(d.Get("data_length").(int))
```

Comment:
Seems SDK use int64 for all int parameters...
I noticed that in the SDK, the comment next to some parameters says that this parameter is of format int32, however in the SDK code the parameter is still defined as an int64.
For example, Revision
```
// The _revision property describes the current revision of the resource. To prevent clients from overwriting each other's changes, PUT operations must include the current _revision of the resource, which clients should obtain by issuing a GET operation. If the _revision provided in a PUT request is missing or stale, the operation will be rejected. format: int32
Revision *int64
```